perceval/backends/core/finosmeetings.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018-2019 Fintech Open Source Foundation
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA.
+#
+# Authors:
+#     Maurizio Pillitu <maoo@finos.org>
+#
+
+import logging
+
+import csv
+import datetime
+import base64
+import urllib.request
+import tempfile
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+
+CATEGORY_ENTRY = "finos-meeting-entry"
+
+logger = logging.getLogger(__name__)
+
+
+class FinosMeetings(Backend):
+    """FinosMeetings backend for Perceval.
+
+    This class retrieves the entries from a CSV file.
+    To initialize this class the CSV file path must be provided.
+    The `file_path` will be set as the origin of the data.
+
+    :param file_path: File Path
+    :param csv_header: Columns included in the CSV file
+    :param separator: CV separator char
+    :param date_formats: Comma-separated list of date formas to use to extract the timestamp of a CSV entry
+    :param skip_header: 'true' if the first CSV row contains the column header
+    :param id_columns: the columns that compose the ID hash
+    :param date_column: the column containing the date for metadata_updated_on
+    """
+    version = '0.0.1'
+
+    CATEGORIES = [CATEGORY_ENTRY]
+
+    separator = ','
+
+    def __init__(self, file_path, csv_header, separator, date_formats, skip_header, id_columns, date_column, tag=None, archive=None):
+        origin = file_path
+
+        super().__init__(origin)
+        self.file_path = file_path
+        self.csv_header = csv_header
+        self.separator = separator
+        self.date_formats = date_formats
+        self.skip_header = (skip_header == 'true')
+        self.id_columns = id_columns
+        self.date_column = date_column
+        self.client = None
+
+    def fetch(self, category=CATEGORY_ENTRY):
+        """Fetch the rows from the CSV.
+
+        :returns: a generator of entries
+        """
+        kwargs = {}
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    @staticmethod
+    def dateToTs(date, formats):
```

Comment:
Please rename this to `date_to_timestamp` or similar. Maybe it can be a module private function because probably nobody is going to use this but this class.